### PR TITLE
Tools: fix lambda test

### PR DESCRIPTION
**Pull request by khancyr**
the lambda checker was called on common.py and not the right files

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -1624,8 +1624,11 @@ def check_test_syntax(self, test_file):
                 # check for lambda: test_function without paranthesis
                 faulty_strings = re.findall(r"lambda\s*:\s*\w+.\w+\s*\)", f.read())
                 if faulty_strings:
-                    desc = ("Syntax error in autotest lambda at : " +
-                            faulty_strings)
+                    desc = "Syntax error in autotest lambda at :"
+                    print(desc)
+                    for x in range(len(faulty_strings)):
+                        desc = faulty_strings[x]
+                        print(desc)
```
Comment (OXINARF) [inline Tools/autotest/common.py]: I agree that we should print the faulty strings, but this now seems a weird way to do things: we print an error message, then throw an exception with that message in details, then catch the exception and print a separate message.

By the way, you are passing only the last faulty string to the Exception, that's clearly a bug.

File: Tools/autotest/common.py [hunk 2]
```diff
@@ -2167,7 +2170,6 @@ def test_skipped(self, test, reason):
 
     def run_tests(self, tests):
         """Autotest vehicle in SITL."""
-        self.check_test_syntax(test_file=os.path.realpath(__file__))
```
Comment (OXINARF) [inline Tools/autotest/common.py]: You forgot to copy this to QuadPlane.

File: Tools/autotest/common.py (hunk 1 above)
Comment (khancyr) [reply Tools/autotest/common.py]: right, corrected

**Comment by khancyr** [comment]
Correction from review

File: Tools/autotest/common.py [hunk 3]
```diff
@@ -1624,18 +1624,20 @@ def check_test_syntax(self, test_file):
...
                     raise ErrorException(desc)
-        except ErrorException:
-            self.progress('FAILED: "%s"' % "Check for syntax mistake in autotest lambda")
+        except ErrorException as msg:
+            self.progress("FAILED: Check for syntax mistake in autotest lambda. \n" + str(msg))
             exit(1)
-        self.progress('PASSED: "%s"' % "Check for syntax mistake in autotest lambda")
+        self.progress("PASSED: Check for syntax mistake in autotest lambda")
 
     @abc.abstractmethod
```
Comment (peterbarker) [inline Tools/autotest/common.py]: This doesn't look particularly abstract.

Comment (khancyr) [reply Tools/autotest/common.py]: fixed

**Comment by khancyr** [comment]
@peterbarker comment fix, ready to go

**Comment by peterbarker** [comment]
Merged, thanks.

**Pull request merged**